src/main/java/TaskList.java

```diff
@@ -0,0 +1,178 @@
+                Todo todo = new Todo(split[1]);
+                lst.add(todo);
+                System.out.println("You got it! I added this task:\n   "
+                        + todo.toString());
+                countTasks();
+            } else {
+                try {
+                    String[] separateDetails = split[1].split("/by |/at ");
+                    String description = separateDetails[0];
+
+                    String date = separateDetails[1];
+                    LocalDate localDate = LocalDate.parse(date);
+
+                    if (command.startsWith(ADD_DEADLINE_COMMAND)) {
+                        Deadline deadline = new Deadline(description, localDate);
+
+                        lst.add(deadline);
+                        System.out.println("You got it! I added this task:\n   "
+                                + deadline.toString());
+                    } else {
+                        // command is ADD_EVENT_COMMAND
+                        Event event = new Event(description, localDate);
+                        lst.add(event);
+                        System.out.println("You got it! I added this task:\n   "
+                                + event.toString());
+                    }
+                    countTasks();
+                } catch (ArrayIndexOutOfBoundsException e){
+                    System.err.println("Did you format your request properly? " +
+                            "This is getting old.");
+                }
+            }
+            Ui.formatText();
+        } catch (DateTimeParseException e) {
+            System.err.println("Your date formatting is invalid, use YYYY-MM-DD please...");
+            Ui.formatText();
+        }
+    }
+
+    /**
+     * Deletes task from list of tasks.
+     * @param str task number given one-based indexing
+     */
+    public void deleteTask(String str) {
+        try {
+            int index = Integer.parseInt(str) - 1;
+            Task task = lst.get(index);
+            lst.remove(index);
+
+            Ui.formatText();
+            System.out.println("Sweet! I have deleted the following task:\n"
+                    + task.toString());
+            countTasks();
+            Ui.formatText();
+        } catch (final NumberFormatException e) {
+            System.err.println("Oof, did you type a valid number or not?");
+        } catch (IndexOutOfBoundsException exception) {
+            System.err.println("You don't have so many items, dumbass!");
+        }
+    }
+
+    public void searchTask(String str) {
+        System.out.println("Here are the items that match your search:");
+        int i = 1;
+        for (Task item : lst) {
+            if (item.description.toLowerCase().contains(str.toLowerCase())) {
+                System.out.println(i + ". " + item.toString());
+                i++;
+            }
+        }
+    }
+
+    /**
+     * Determines if task is Todo, Deadline or Event and handles them accordingly.
```

Comment:
Do you think it would be clearer to explain that this method is only called when read from storage?